File: content/search/source/includes/synonyms-tutorial/code-snippets/nodejs/multiple-synonyms.js
const { MongoClient } = require('mongodb');

async function createMultipleSynonyms() {
  // Connection URI
  const uri = '<connection-string>';
  
  // Create a new MongoClient
  const client = new MongoClient(uri);

  try {
    // Connect to the MongoDB server
    await client.connect();
    
    // Get the sample_mflix database
    const database = client.db('sample_mflix');
    
    // Create the transport_synonyms collection
    try {
      await database.createCollection('transport_synonyms');
    } catch (err) {
      // Collection may already exist, which is fine
      console.log(`Note: ${err.message}`);
    }
    
    // Get the transport_synonyms collection
    const transportCollection = database.collection('transport_synonyms');
    
    // Create and insert the first transport document - equivalent mapping
    const doc1 = {
      mappingType: 'equivalent',
      synonyms: ['car', 'vehicle', 'automobile']
    };
    
    await transportCollection.insertOne(doc1);
    
    // Create and insert the second transport document - explicit mapping
    const doc2 = {
      mappingType: 'explicit',
      input: ['boat'],
      synonyms: ['boat', 'vessel', 'sail']
    };
    
    await transportCollection.insertOne(doc2);
    
    // Create the attire_synonyms collection
    try {
      await database.createCollection('attire_synonyms');
    } catch (err) {
      // Collection may already exist, which is fine
      console.log(`Note: ${err.message}`);
    }
    
    // Get the attire_synonyms collection
    const attireCollection = database.collection('attire_synonyms');
    
    // Create and insert the first attire document - equivalent mapping
    const doc3 = {
      mappingType: 'equivalent',
      synonyms: ['dress', 'apparel', 'attire']
    };
    
    await attireCollection.insertOne(doc3);
    
    // Create and insert the second attire document - explicit mapping
    const doc4 = {
      mappingType: 'explicit',
      input: ['hat'],
      synonyms: ['hat', 'fedora', 'headgear']
    };
    
    await attireCollection.insertOne(doc4);
    
    console.log('Synonyms collections successfully created and populated.');
  } catch (err) {
    console.error(`Error: ${err.message}`);
    process.exit(1);
  } finally {
    // Close the connection
    await client.close();
  }
}

// Run the function and handle any errors
createMultipleSynonyms().catch(console.error);

File: content/search/source/includes/synonyms-tutorial/code-snippets/csharp/MultipleSynonyms.cs
using System;
using MongoDB.Bson;
using MongoDB.Driver;

namespace SynonymsTutorial
{
    class Program
    {
        static void Main(string[] args)
        {
            try
            {
                // Connection string to your MongoDB cluster
                string connectionString = "<connection-string>";
                
                // Create a MongoDB client
                var client = new MongoClient(connectionString);
                
                // Get the sample_mflix database
                var database = client.GetDatabase("sample_mflix");
                
                // Create the transport_synonyms collection
                try
                {
                    database.CreateCollection("transport_synonyms");
                }
                catch (MongoCommandException ex)
                {
                    // Collection may already exist, which is fine
                    Console.WriteLine($"Note: {ex.Message}");
                }
                
                var transportCollection = database.GetCollection<BsonDocument>("transport_synonyms");
                
                // Create and insert the first transport document - equivalent mapping
                var doc1 = new BsonDocument
                {
                    { "mappingType", "equivalent" },
                    { "synonyms", new BsonArray { "car", "vehicle", "automobile" } }
                };
                
                transportCollection.InsertOne(doc1);
                
                // Create and insert the second transport document - explicit mapping
                var doc2 = new BsonDocument
                {
                    { "mappingType", "explicit" },
                    { "input", new BsonArray { "boat" } },
                    { "synonyms", new BsonArray { "boat", "vessel", "sail" } }
                };
                
                transportCollection.InsertOne(doc2);
                
                // Create the attire_synonyms collection
                try
                {
                    database.CreateCollection("attire_synonyms");
                }
                catch (MongoCommandException ex)
                {
                    // Collection may already exist, which is fine
                    Console.WriteLine($"Note: {ex.Message}");
                }
                
                var attireCollection = database.GetCollection<BsonDocument>("attire_synonyms");
                
                // Create and insert the first attire document - equivalent mapping
                var doc3 = new BsonDocument
                {
                    { "mappingType", "equivalent" },
                    { "synonyms", new BsonArray { "dress", "apparel", "attire" } }
                };
                
                attireCollection.InsertOne(doc3);
                
                // Create and insert the second attire document - explicit mapping
                var doc4 = new BsonDocument
                {
                    { "mappingType", "explicit" },
                    { "input", new BsonArray { "hat" } },
                    { "synonyms", new BsonArray { "hat", "fedora", "headgear" } }
                };
                
                attireCollection.InsertOne(doc4);
                
                Console.WriteLine("Synonyms collections successfully created and populated.");
            }
            catch (Exception ex)
            {
                Console.WriteLine($"Error: {ex.Message}");
                Environment.Exit(1);
            }
        }
    }
}
